offchain/danaswap-browser/src/DanaSwap/Browser/Home.purs

```diff
@@ -19,18 +24,239 @@ main = HA.runHalogenAff do
+  handleAction = case _ of
+    ResetCurrentPool -> H.modify_ _ { currentPool = Nothing }
+    SetCurrentPool pool -> H.modify_ _ { currentPool = Just pool }
+    SetActivity activity -> H.modify_ _ { activity = activity }
+
+  render :: forall slots. State -> H.ComponentHTML Action slots m
+  render { pools, currentPool, activity } =
+    HH.section
+      [ mkClass "section my-6 pb-6" ]
+      $
+        [ HH.div
+            [ mkClass "columns is-mobile" ]
+            [ HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-left" ] [ HH.text "Pool" ]
+                    ]
+                ]
+            , HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-centered" ] [ HH.text "Value" ]
+                    ]
+                ]
+            , HH.div [ mkClass "pools-table-header column" ]
+                [ HH.div [ mkClass $ "columns" ]
+                    [ HH.div [ mkClass "column has-text-centered" ] [ HH.text "Fee" ]
+                    ]
+                ]
+            , HH.div [ mkClass "column" ] [ HH.text " " ]
+            ]
+
+        ]
+      <> concat
+        ( renderPool
+            <$> pools
+        )
+    where
+    renderPool pool@(Pool p) =
+      [ HH.div
+          [ mkClass "pools-table-row columns is-mobile is-vcentered p-3 my-3"
+          , HE.onClick $ \_ ->
+              if currentPool == Just pool then
+                ResetCurrentPool
+              else
+                SetCurrentPool pool
+          ]
+
+          [ HH.div [ mkClass "column" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch has-text-left" ] [ HH.text $ show pool ]
+              ]
+          , HH.div [ mkClass "column" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch has-text-centered" ] [ HH.text $ show p.assetA <> " / " <> show p.assetB ]
+              ]
+          , HH.div [ mkClass "column" ]
+              [ HH.p [ mkClass "is-size-5-desktop is-size-7-touch has-text-centered" ] [ HH.text $ show p.fee <> "ADA" ]
+              ]
+          , HH.div [ mkClass "column has-text-centered" ]
+              [ if currentPool == Just pool then mkIcon "angle-up" else mkIcon "angle-down"
+              ]
+          ]
+      ] <> renderPoolDropdown pool
+
+    renderPoolDropdown pool =
+      if currentPool == Just pool then
+        [ HH.div [ mkClass "pools-table-row-dropdown box my-6" ]
+            [ HH.div [ mkClass "columns is-mobile" ]
+                [ HH.div [ mkClass "column is-narrow" ]
+                    [ HH.button [ HE.onClick $ const (SetActivity Swap), mkClass $ "button is-rounded is-medium is-responsive " <> if (activity == Swap) then "danaswap-btn-has-background" else "danaswap-btn-has-border" ] [ HH.text "SWAP" ]
```

Comment:
Use CSS `text-transform: uppercase` to UPPERCASE your button text

Reply:
I am curious what's advantage of using CSS instead of plain text?

Reply:
1. You're not communicating what you mean to machines, because this is purely a visual choice
2. Because it's just presentation, you've taken something that's reversible, to something that isn't. If the design later calls for a different case, you've lost that information. A good example is "iPhone": if I put "IPHONE" in here, there is no reasonable way to get the original casing back. If you say `titleCase "IPHONE"`, you get "Iphone".

If you do this at the presentation layer, CSS, then you can communicate the original casing to machines and have a casing that's reversible. Do it make sense that you're not communicating "SWAP" but you're choosing to present "Swap" as "SWAP" for aesthetic reasons?